pkg/issuer/acme/dns/route53/route53.go

```diff
@@ -53,43 +53,33 @@ func (d customRetryer) RetryRules(r *request.Request) time.Duration {
 }
 
 // NewDNSProvider returns a DNSProvider instance configured for the AWS
-// Route 53 service.
-//
-// AWS Credentials are automatically detected in the following locations
-// and prioritized in the following order:
-// 1. Environment variables: AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY,
-//    AWS_REGION, [AWS_SESSION_TOKEN]
-// 2. Shared credentials file (defaults to ~/.aws/credentials)
-// 3. Amazon EC2 IAM role
-//
-// If AWS_HOSTED_ZONE_ID is not set, Lego tries to determine the correct
-// public hosted zone via the FQDN.
-//
-// See also: https://github.com/aws/aws-sdk-go/wiki/configuring-sdk
-func NewDNSProvider() (*DNSProvider, error) {
-	hostedZoneID := os.Getenv("AWS_HOSTED_ZONE_ID")
-
-	r := customRetryer{}
-	r.NumMaxRetries = maxRetries
-	config := request.WithRetryer(aws.NewConfig(), r)
-	client := route53.New(session.New(config))
-
-	return &DNSProvider{
-		client:       client,
-		hostedZoneID: hostedZoneID,
-	}, nil
-}
-
-// NewDNSProviderAccessKey returns a DNSProvider instance configured for the AWS
-// Route 53 service using static credentials from its parameters
-func NewDNSProviderAccessKey(accessKeyID, secretAccessKey, hostedZoneID, region string) (*DNSProvider, error) {
+// Route 53 service using static credentials from its parameters or, if they're
+// unset and the 'ambient' option is set, credentials from the environment.
+func NewDNSProvider(accessKeyID, secretAccessKey, hostedZoneID, region string, ambient bool) (*DNSProvider, error) {
+	if accessKeyID == "" && secretAccessKey == "" {
+		if !ambient {
+			return nil, fmt.Errorf("unable to construct route53 provider: empty credentials; perhaps you meant to enable ambient credentials?")
+		}
+	} else if accessKeyID == "" || secretAccessKey == "" {
+		// It's always an error to set one of those but not the other
+		return nil, fmt.Errorf("unable to construct route53 provider: only one of access and secret key was provided")
+	}
 
-	creds := credentials.NewStaticCredentials(accessKeyID, secretAccessKey, "")
+	useAmbientCredentials := ambient && (accessKeyID == "" && secretAccessKey == "")
 
 	r := customRetryer{}
 	r.NumMaxRetries = maxRetries
+	config := request.WithRetryer(aws.NewConfig(), r)
 
-	config := request.WithRetryer(aws.NewConfig(), r).WithCredentials(creds)
+	if useAmbientCredentials {
+		glog.V(5).Infof("using ambient credentials")
+		// Leaving credentials unset results in a default credential chain being
+		// used; this chain is a reasonable default for getting ambient creds.
+		// https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
+	} else {
+		glog.V(5).Infof("not using ambient credentials")
+		config.WithCredentials(credentials.NewStaticCredentials(accessKeyID, secretAccessKey, ""))
```

Comment:
If these credentials are invalid, will the default chain still be used? (i.e. does WithCredentials append to the chain, or replace?)